lib/src/detectors/virtualbox_detector.cc

```diff
@@ -0,0 +1,28 @@
+#include <internal/detectors/virtualbox_detector.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <leatherman/execution/execution.hpp>
+#include <leatherman/util/regex.hpp>
+#include <unordered_map>
+
+using namespace std;
+using namespace whereami;
+using namespace leatherman::execution;
+using namespace leatherman::util;
+
+namespace whereami { namespace detectors {
+
+    /**
+     * Map of VM names to regex patterns returned by data sources
+     */
+    static const unordered_map<string, boost::regex> vm_patterns {
```

Comment:
Since each detector is now responsible on its own, I think we can get rid of the overhead of this, and just create the regex inline in the detector function.